src/modules/GroupChannel/context/hooks/useMessageActions.ts

```diff
@@ -0,0 +1,144 @@
+import { useGroupChannelMessages } from '@sendbird/uikit-tools';
+import type { SendableMessageType } from '../../../../utils';
+import { ReplyType } from '../../../../types';
+import { useCallback } from 'react';
+import {
+  BaseMessageCreateParams,
+  FileMessage,
+  FileMessageCreateParams,
+  MessageMetaArray,
+  MultipleFilesMessage,
+  MultipleFilesMessageCreateParams,
+  UserMessage,
+  UserMessageCreateParams,
+  UserMessageUpdateParams,
+} from '@sendbird/chat/message';
+import {
+  META_ARRAY_MESSAGE_TYPE_KEY,
+  META_ARRAY_MESSAGE_TYPE_VALUE__VOICE,
+  META_ARRAY_VOICE_DURATION_KEY,
+  VOICE_MESSAGE_FILE_NAME,
+  VOICE_MESSAGE_MIME_TYPE,
+} from '../../../../utils/consts';
+import { GroupChannelProviderProps } from '../GroupChannelProvider';
+
+type MessageListDataSource = ReturnType<typeof useGroupChannelMessages>;
+type MessageActions = {
+  sendUserMessage: (params: UserMessageCreateParams) => Promise<UserMessage>;
+  sendFileMessage: (params: FileMessageCreateParams) => Promise<FileMessage>;
+  sendVoiceMessage: (params: FileMessageCreateParams, duration: number) => Promise<FileMessage>;
+  sendMultipleFilesMessage: (params: MultipleFilesMessageCreateParams) => Promise<MultipleFilesMessage>;
+  updateUserMessage: (messageId: number, params: UserMessageUpdateParams) => Promise<UserMessage>;
+};
+
+interface Params extends GroupChannelProviderProps, MessageListDataSource {
+  scrollToBottom(): Promise<void>;
+  quoteMessage?: SendableMessageType;
+  replyType: ReplyType;
+}
+
+const pass = <T>(value: T) => value;
+
+/**
+ * @description This hook controls common processes related to message sending, updating.
+ * */
+export function useMessageActions(params: Params): MessageActions {
+  const {
+    onBeforeSendUserMessage = pass,
+    onBeforeSendFileMessage = pass,
+    onBeforeUpdateUserMessage = pass,
+    onBeforeSendVoiceMessage = pass,
+    onBeforeSendMultipleFilesMessage = pass,
+
+    sendFileMessage,
+    sendMultipleFilesMessage,
+    sendUserMessage,
+    updateUserMessage,
+
+    scrollToBottom,
+    quoteMessage,
+    replyType,
+  } = params;
+
+  const buildInternalMessageParams = useCallback(
+    <T extends BaseMessageCreateParams>(basicParams: T): T => {
+      const messageParams = { ...basicParams } as T;
+
+      if (params.quoteMessage && replyType !== 'NONE') {
+        messageParams.isReplyToChannel = true;
+        messageParams.parentMessageId = quoteMessage.messageId;
+      }
+
+      // TODO: check isMentionEnabled should be handled here.
```

Comment:
The properties related with mention feature shouldn't be removed, if it's used for message updating